syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.freqscrap";
option java_outer_classname = "FreqScrapProto";
option objc_class_prefix = "FS";

package freqscrap;

// Interface exported by the server.
service FreqScrap {
  // A simple RPC.
  //
  // Obtains the feature at a given position.
  //
  // A feature with an empty name is returned if there's no feature at the given
  // position.
  rpc GetFreq(Request) returns (Response) {}

}

// Points are represented as latitude-longitude pairs in the E7 representation
// (degrees multiplied by 10**7 and rounded to the nearest integer).
// Latitudes should be in the range +/- 90 degrees and longitude should be in
// the range +/- 180 degrees (inclusive).
message Request {
  string url = 1;
  string scrap_domains = 2;
  int32 ngrams = 3;
  int32 top = 4;
  int32 max_level = 5;
}

message Response {
  repeated Result results = 1;
}

message Result {
  string word = 1;
  int32 freq = 2;
}